core/store/migrations/migration1605816413/migrate.go

```diff
@@ -0,0 +1,15 @@
+package migration1605816413
+
+import "github.com/jinzhu/gorm"
+
+// Note this destroys the ability to have an FK on jobs_specs(LOWER(name)),
+// however that's ok given we have serial PK on the table which can be used for FKs.
+// The second unique partial index ensures we cannot have multiple jobs names with the same
+// name but different, not null deleted_at timestamps.
+func Migrate(tx *gorm.DB) error {
+	return tx.Exec(`
+        DROP INDEX job_specs_name_index;
+        CREATE UNIQUE INDEX job_specs_name_index_active on job_specs (LOWER(name)) where deleted_at is NULL;
+        CREATE UNIQUE INDEX job_specs_name_index_deleted on job_specs (LOWER(name)) where deleted_at is NOT NULL;
+    `).Error
```

Comment:
Don't we want the ability to have multiple jobs with the same name, but different deleted_at timestamps? I want to be able to soft delete jobs as many times as I need without worrying about name collisions, right?

Reply:
yep - you're right

Reply:
Good catch @RyanRHall